artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/federation/AMQPFederation.java

```diff
@@ -239,7 +243,7 @@ public synchronized AMQPFederation addQueueMatchPolicy(FederationReceiveFromQueu
     * @throws ActiveMQException if an error occurs processing the added policy
     */
    public synchronized AMQPFederation addAddressMatchPolicy(FederationReceiveFromAddressPolicy addressPolicy) throws ActiveMQException {
-      final FederationAddressPolicyManager manager = new AMQPFederationAddressPolicyManager(this, addressPolicy);
+      final FederationAddressPolicyManager manager = new AMQPFederationAddressPolicyManager(this, addressPolicy, server.getConfiguration().isLargeMessageSync());
```

Comment:
Adding unneeded constructor args makes the code more unreadable and less maintainable.  The bits that need to inspect this value can simply ask the connection or session SPI for the value when it is needed.

Reply:
@tabish121 I don't see an alternative here. Federation has no access to the SPI.

I could add the SPI, other stuff... but I need to add something.

Reply:
You don't need it, that's the issue, the places it is use all have access to ProtonAbstractReceiver which can call, getSessionContext() on it and ask if sync is needed